src/MiniCover.HitServices/HitTestMethod.cs

```diff
@@ -76,6 +76,7 @@ public static IEnumerable<HitTestMethod> MergeDuplicates(IEnumerable<HitTestMeth
 
         public void Serialize(Stream stream)
         {
+            lock (_lock)
```

Comment:
_lock is scoped to an instance of HitServiceMethod, you should lock stream instead.
Also, check if there is no lock on stream already on the class that calls this one.